driver-core/src/main/com/mongodb/connection/CommandMessage.java

```diff
@@ -48,6 +49,8 @@
     private final SplittablePayload payload;
     private final FieldNameValidator payloadFieldNameValidator;
     private final boolean responseExpected;
+    private boolean messageHasBeenEncoded;
```

Comment:
You can ditch this boolean in favor of getEncodingMetadata() != null

Reply:
Done.